tck/src/main/java/org/eclipse/microprofile/config/tck/PropertyExpressionsTest.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2016-2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.config.tck;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+public class PropertyExpressionsTest {
+    @Test
```

Comment:
This is unit test. How can we test against the runtime to ensure the runtime support this? We need to use Arqullian way to package an app though.

Reply:
If you look here:
https://github.com/eclipse/microprofile-config/pull/577#discussion_r463547817

I just create Programatic Config. The Config Provider is available when running the TCK, so it does work. And since we don't need CDI or other features from Arquillian, we don't really need to packaged this in an app.

Reply:
BTW, I'm able to run the test just fine in SR Config TCK.

Reply:
I took a further look at the tests. You do need to test CDI, which is why I asked.

`@Inject @ConfigProperty(name="url") SomeType t;`
By the way, please also add another test to ensure the defaultValue can contain ${}, which will be expanded as well.